### ISSUE TITLE
ML training on strava activities as examples (Learning Process)

### ISSUE THREAD

**Issue by eaudet** [eaudet/trainingbot#11]
![image](https://user-images.githubusercontent.com/288285/40340325-66372414-5d4c-11e8-93c7-4f80f4acb9dd.png)

**Comment by eaudet** [comment]
Prediction using regression: https://spark.apache.org/examples.html

**Comment by eaudet** [comment]
https://spark.apache.org/docs/latest/mllib-linear-methods.html#linear-least-squares-lasso-and-ridge-regression

**Comment by eaudet** [comment]
https://spark.apache.org/docs/latest/ml-classification-regression.html#multinomial-logistic-regression

**Comment by eaudet** [comment]
https://dzone.com/articles/predictive-analytics-with-spark-ml

**Comment by eaudet** [comment]
**UPTO**:
To build and train the classification model, we have a createPipelineModel method below. Comments provided in the method describe what is being done, but essentially, we are “featurizing” our training data, creating a single stage pipeline to run multiple random forest hyperparameter combinations, and running cross validation to find the best predictive model:

**Comment by eaudet** [comment]
http://waikato.github.io/meka/

**Comment by eaudet** [comment]
https://github.com/tensorflow/tensorflow/tree/r1.8/tensorflow/java

**Comment by eaudet** [comment]
Based on plan sessions create variations of plans (under, normal and over). This will become the training sets. Must validate ATL, CTL, TSB are showing valid values.

**Comment by eaudet** [comment]
Cannot get the strava power (bike) to calculate NP.

**Comment by eaudet** [comment]
https://developers.strava.com/playground/

**Comment by eaudet** [comment]
No generating bike actual ftp reflecting under or over power.

**Comment by eaudet** [comment]
Generator must create power range data based on over/under/normal config. Now they are all the same.

**Comment by eaudet** [comment]
Trainingset (raw data seems ok). Now, I need to change (adapt) features based on the excel sheet example.

**Comment by eaudet** [comment]
Need to update session and activity generation using this spreadsheet: https://docs.google.com/spreadsheets/d/1JM1dwrjPzQ1D0GrWGCtM_nQ-9--mrMZ9xUMAQsDJxo8/edit?usp=sharing

**Comment by eaudet** [comment]
Todo: Add feature normalization
X'= (X - Xmin) / (Xmax - Xmin)

**Comment by eaudet** [comment]
How do I save the model?

**Comment by eaudet** [comment]
Model is generated and serialized. Spring service is tested.
